### PR TITLE
Use "major version" and "minor version" for syntactic versions

### PR DESCRIPTION
**Pull request by ishanjainn**
This PR makes to use `major` and `minor` version instead of Sequence and Schema versions naming where appropriate. Variables and fields are renamed from `seqv` and `schv` to `majv` and `minv`

Resolves https://github.com/grafana/thema/issues/54

### CODE REVIEW

**Comment by ishanjainn** [comment]
@sdboyer 
I have updated the docs but wanted to take your opinion if the docs part makes the version part clearer?

File: docs/go-usage.md [hunk 1]
```diff
@@ -121,7 +121,7 @@ func TestHandpickValidation(t *testing.T) {
 }
 ```
 
-Here, we've hand-picked the schema version we want to validate against - `0.0`, which every lineage is guaranteed to contain. The [`LatestVersion()`](https://pkg.go.dev/github.com/grafana/thema#LatestVersion) and [`LatestVersionInSequence()`](https://pkg.go.dev/github.com/grafana/thema#LatestVersionInSequence) functions provide fuzzier version selection logic. But allowing only one schema version as input somewhat defeats the purpose of using Thema in the first place. Ideally, we'd have something more dynamic.
+Here, we've hand-picked the syntactic version of the schema we want to validate against - `0.0`, which every lineage is guaranteed to contain. The [`LatestVersion()`](https://pkg.go.dev/github.com/grafana/thema#LatestVersion) and [`LatestVersionInSequence()`](https://pkg.go.dev/github.com/grafana/thema#LatestVersionInSequence) functions provide fuzzier version selection logic. But allowing only one syntactic version of a schema from the Lineage of theas input somewhat defeats the purpose of using Thema in the first place. Ideally, we'd have something more dynamic.
```
Comment (sdboyer) [inline docs/go-usage.md:124]:
```suggestion
Here, we've hand-picked the syntactic version of the schema we want to validate against - `0.0`, which every lineage is guaranteed to contain. The [`LatestVersion()`](https://pkg.go.dev/github.com/grafana/thema#LatestVersion) and [`LatestVersionInSequence()`](https://pkg.go.dev/github.com/grafana/thema#LatestVersionInSequence) functions provide fuzzier version selection logic. But allowing only one syntactic version as input somewhat defeats the purpose of using Thema in the first place. Ideally, we'd have something more dynamic.
```

**Review by sdboyer — COMMENTED** [review]
A number of these changes make explicit that when the docs are talking about version numbers, it's a syntactic version, of a schema, in a lineage.

I'm undecided about this. When i wrote the docs originally, i thought, "it's preferable to assume that the reader has understood that all versions in thema are syntactic versions, and thus write shorter, more readable sentences that leave that implicit."

However, i also recognized that i'm a terrible judge of whether that's a reasonable assumption or not, because i have the [curse of knowledge](https://en.wikipedia.org/wiki/Curse_of_knowledge) here. So i'm happy to defer - do you find it preferable to be more verbose, explicitly stating that versions are syntactic versions?

**Comment by ishanjainn** [comment]
Wasnt in the favour of being more verbose, Instead Ill prefer just explain as a note within the first doc the user reads(overview.md) what `schema version` means and continue using it thereafter. Let me

**Pull request closed without merging**